sklearn/metrics/classification.py

```diff
@@ -1225,14 +1250,45 @@ class 2       1.00      0.67      0.80         3
     else:
         labels = np.asarray(labels)
 
-    last_line_heading = 'avg / total'
+    is_binary_classification = len(labels) == 2
+    is_multilabel = _check_targets(y_true, y_pred)[0].startswith('multilabel')
+
+    if is_binary_classification and average:
+        raise ValueError('Averaging methods are not relevant for binary '
+                         'classification. Please omit average argument.')
```

Comment:
I disagree, passing an averaging method explicitly makes no sense for the binary classification case.

Still it would be cleaner to write:

```python
if is_binary_classification and average is not None:
    ...
```